python/ray/train/tests/test_backend.py

```diff
@@ -178,6 +191,21 @@ def train_func():
         assert list(e.finish_training()) == [2, 2, 1]
 
 
+def test_local_world_size_with_same_ip_nodes(ray_2_node_2_cpu):
+    config = TestConfig()
+    with patch.object(
+        WorkerGroup, "add_workers", mock_add_workers_to_nodes_with_same_ip
```

Comment:
The 2 virtual nodes will have the same node_ip. Can we also have a test without patching the `add_worker` method?